### PR TITLE
Change Guava version to 31 and Flink to 1.18

### PR DESCRIPTION
**Pull request by maver1ck**
Upgrade Guava to version 31 to get compatibility with Flink 1.18

### CODE REVIEW

**Comment by gtk96** [comment]
@maver1ck flink 1.18 already released

**Comment by maver1ck** [comment]
@gtk96 PR rebased, let's check tests

**Comment by maver1ck** [comment]
@gtk96 
Any idea why those tests are failing ? Looks like timeout not related to PR.

**Comment by gtk96** [comment]
> @gtk96 Any idea why those tests are failing ? Looks like timeout not related to PR.

I read the log. I think you are right. There are other reasons for the error. cc @GOODBOY008 @ruanhang1993

**Comment by Kakaeex** [comment]
Look forward to your good news. when upgrade to jdk17，so many Plug-ins need to upgrade

**Comment by maver1ck** [comment]
@Kakaeex Were you able to compile this repo in Java 17 ? I'm using still Java 11 and Java 17 only for Flink jobs.

**Comment by Jiabao-Sun** [comment]
Thanks @maver1ck for this great work and sorry for the late review.
Would you mind to resolve the conflicts?

We can refer to the PR that bumped version to 1.17 #2057.

**Comment by maver1ck** [comment]
@Jiabao-Sun 
I rebased this PR.

**Comment by Jiabao-Sun** [comment]
The CI fails, it seems that we have ignored some newly added classes.

By the way, can we add version 1.18 in the E2E test?
https://github.com/ververica/flink-cdc-connectors/blob/80dfc515a1ec26c3a1d46aa6cbc7264d0ff3a8c9/flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java#L125-L131

https://github.com/ververica/flink-cdc-connectors/blob/80dfc515a1ec26c3a1d46aa6cbc7264d0ff3a8c9/flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/pom.xml#L28-L36

File: pom.xml [hunk 1]
```diff
@@ -152,7 +152,7 @@ under the License.
         <dependency>
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-shaded-guava</artifactId>
-            <version>30.1.1-jre-16.1</version>
+            <version>31.1-jre-17.0</version>
```
Comment (Jiabao-Sun) [inline pom.xml:155]: I think `<slf4j.version>` can be changed to 1.7.36 as well.

**Comment by maver1ck** [comment]
@Jiabao-Sun will do tomorrow.

**Comment by maver1ck** [comment]
OK. I think we have missing dependency on jdbc-connector_1.18.0.jar.
AFAIK there is no version of JDBC connector for 1.18.0. Let me add PR in their repo.

**Comment by maver1ck** [comment]
We are blocked by https://github.com/apache/flink-connector-jdbc/pull/76

File: flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/pom.xml [hunk 2]
```diff
@@ -30,7 +30,9 @@ under the License.
         <flink-1.15>1.15.2</flink-1.15>
         <flink-1.16>1.16.0</flink-1.16>
         <flink-1.17>1.17.0</flink-1.17>
-        <jdbc.version-1.17>3.1.0-1.17</jdbc.version-1.17>
+        <flink-1.18>1.18.0</flink-1.18>
```
Comment (gtk96) [inline flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/pom.xml:33]: 1.16.2
1.17.1   WDYT?

File: flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java [hunk 3]
```diff
@@ -124,11 +124,11 @@ public abstract class FlinkContainerTestEnvironment extends TestLogger {
 
     @Parameterized.Parameters(name = "flinkVersion: {0}")
     public static List<String> getFlinkVersion() {
-        return Arrays.asList("1.14.4", "1.15.2", "1.16.0", "1.17.0");
+        return Arrays.asList("1.14.4", "1.15.2", "1.16.0", "1.17.0", "1.18.0");
     }
```
Comment (gtk96) [inline flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java:128]: 1.16.2
1.17.1   WDYT?

File: flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java [hunk 4]
```diff
@@ -124,11 +124,11 @@ public abstract class FlinkContainerTestEnvironment extends TestLogger {
...
 
     private static final List<String> FLINK_VERSION_WITH_SCALA_212 =
-            Arrays.asList("1.15.2", "1.16.0", "1.17.0");
+            Arrays.asList("1.15.2", "1.16.0", "1.17.0", "1.18.0");
```
Comment (gtk96) [inline flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java:131]: 1.16.2
1.17.1   WDYT?

Comment (maver1ck) [reply flink-cdc-e2e-tests/flink-cdc-source-e2e-tests/src/test/java/com/ververica/cdc/connectors/tests/utils/FlinkContainerTestEnvironment.java]: Fixed.

**Review by leonardBang — APPROVED** [review]
Thanks @maver1ck for the contribution and @Jiabao-Sun for the review work, LGTM

**Pull request merged**